tools/Azure.Mcp.Tools.ApplicationInsights/src/Commands/Recommendation/RecommendationListCommand.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.Text.Json.Nodes;
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Commands.Subscription;
+using Azure.Mcp.Core.Models.Command;
+using Azure.Mcp.Core.Services.Telemetry;
+using Azure.Mcp.Tools.ApplicationInsights.Models;
+using Azure.Mcp.Tools.ApplicationInsights.Options;
+using Azure.Mcp.Tools.ApplicationInsights.Services;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.ApplicationInsights.Commands.Recommendation;
+
+public sealed class RecommendationListCommand(ILogger<RecommendationListCommand> logger) : SubscriptionCommand<RecommendationListOptions>()
+{
+    private const string CommandTitle = "List Application Insights Recommendations";
+    private readonly ILogger<RecommendationListCommand> _logger = logger;
+
+    public override string Name => "list";
+
+    public override string Description =>
+        """
+        List Application Insights Code Optimization Recommendations in a subscription. Optionally filter by resource group when --resource-group is provided.
+        Returns the code optimization recommendations based on the profiler data.
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new() { Destructive = false, ReadOnly = true };
```

Comment:
Please provide the values for all hints.

Reply:
Sure.

Reply:
All values provided. Thanks.